### PR TITLE
Merge 15.0 code freeze into trunk

### PR DESCRIPTION
**Pull request by jkmassel**
Includes:
- Version number bumps
- Release notes
- Production dependencies
- Translation strings

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -95,7 +95,7 @@ tasks.register("installGitHooks", Copy) {
 }
 
 ext {
-    fluxCVersion = 'trunk-f8d0bcf75c60827ac5df99d42a5a85bb95336c2c'
+    fluxCVersion = '2.42.0'
     wooCommerceSharedVersion = 'trunk-09b36638186095b54a31aa3bdedac9d39e5438da'
```
Comment (mokagio) [inline build.gradle:99]: For reference, this commit is [that of the 0.4.0 release](https://github.com/woocommerce/WooCommerce-Shared/commit/09b36638186095b54a31aa3bdedac9d39e5438da).

The latest tag pushed to the remote is [0.6.0](https://github.com/woocommerce/WooCommerce-Shared/releases/tag/0.6.0).

I didn't look into the timeline, but it looks like we now have [support for CI publishing builds from tags](https://buildkite.com/automattic/woocommerce-shared/builds/403#018a08f2-f704-4d3f-ba50-510df1470f96).

Should we start checking and updating the version of this library, too, or is it still too early in development and better left in an ad hoc manual process? cc @Ecarrion @wzieba

**Review by mokagio — APPROVED** [review]
- [x] New version header in `RELEASE-NOTE.txt` 
- [x] Version updates in `build.gradle`
- [x] `release_notes.txt` updated with notes from `RELEASE-NOTE.txt` for current version 
- [x] `strings.xml` updates via automation merging values from libraries
- [x] Switched `ext` versions to stable releases

File: build.gradle (hunk 1 above)
Comment (wzieba) [reply build.gradle:99]: We meant to introduce 0.6.0 in this release, but we missed the deadline - we'll do this in 15.1.

> Should we start checking and updating the version of this library, too, or is it still too early in development and better left in an ad hoc manual process?

Good question @mokagio ! For now, the experiment is paused, so we shouldn't be bothered by it. If the experiment will be resumed, I think it'll be good to left it as manual process, as the tempo of the development is not that high, and we'd like to control what exactly we release 👍 .

Comment (mokagio) [reply build.gradle:99]: Noted. Thanks!

**Pull request merged**